src/switchroot/ostree-prepare-root.c

```diff
@@ -174,60 +235,48 @@ main(int argc, char *argv[])
   if (mount (NULL, "/", NULL, MS_REC|MS_PRIVATE, NULL) < 0)
     {
       perrorv ("Failed to make \"/\" private mount: %m");
-      exit (EXIT_FAILURE);
+      die ();
     }
 
   /* Make deploy_path a bind mount, so we can move it later */
-  if (mount (deploy_path, newroot, NULL, MS_BIND, NULL) < 0)
+  if (mount (deploy_path, deploy_path, NULL, MS_BIND, NULL) < 0)
     {
       perrorv ("failed to initial bind mount %s", deploy_path);
```

Comment:
Not introduced by this change, but:

"failed to initial bind mount" => "failed to create initial bind mount" ?